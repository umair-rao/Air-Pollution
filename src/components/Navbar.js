import { FaArrowLeft } from 'react-icons/fa';
import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Navbar.css';

// eslint-disable-next-line arrow-body-style
const Navbar = () => {
  return (
    <div className="nav-container">
      <a href="/" className="items-center">
        <div className="back-icon">
          <FaArrowLeft />
        </div>
        <div className="nav-title">
          <h1>Air Poluttion</h1>
        </div>
      </a>
      <div className="nav-links">
        <ul>
          <li><Link to="/">Asia Capitals</Link></li>
          <li><Link to="/Africa Capitals">Africa Capitals</Link></li>
          <li>North America Capitals</li>
          <li>South America Capitals</li>
          <li>Europe Capitals</li>
          <li>Australia Capitals</li>
        </ul>
      </div>
    </div>
  );
};

export default Navbar;
