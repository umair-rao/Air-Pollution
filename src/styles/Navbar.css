* {
  margin: 0;
  padding: 0;
}

.nav-container {
  display: flex;
  align-items: center;
  position: sticky;
  top: 0.1px;
  width: 100%;
  margin-bottom: 20px;
  background-color: #2c3531;
  color: #ffeb3b;
}

.nav-container a {
  margin-right: 3rem;
  margin-left: 1.5rem;
  color: #ffeb3b;
  font-size: 1.2rem;
  text-decoration: none;
}

.back-icon {
  margin-right: 3rem;
  margin-left: 1.5rem;
  font-size: 3rem;
  color: #ffeb3b;
}

.time-date {
  font-weight: bold;
}
