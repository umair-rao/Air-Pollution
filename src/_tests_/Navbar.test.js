import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import store from '../Redux/configureStore';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
