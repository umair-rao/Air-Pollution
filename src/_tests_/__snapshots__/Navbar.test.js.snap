// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<div
  className="nav-container"
>
  <a
    className="items-center"
    href="/"
  >
    <div
      className="back-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
        />
      </svg>
    </div>
    <div
      className="nav-heading"
    >
      <h1
        className="nav-title"
      >
        Worldwide COVID-19 Live Data Analysis
      </h1>
    </div>
  </a>
  <div
    className="time-date"
  >
    <div
      className="time"
    >
      <p>
        Time:
         
        2:19:11 pm
      </p>
    </div>
    <div
      className="date"
    >
      <p>
        Date:
         
        14-1-2023
      </p>
    </div>
  </div>
</div>
`;
